Navigation2 build failure on Ubuntu 16.04

<!--
For general questions, please ask on ROS answers: https://answers.ros.org, make sure to include at least the `ros2` tag and the rosdistro version you are running, e.g. `ardent`.
For general design discussions, please post on discourse: https://discourse.ros.org/c/ng-ros
Not sure if this is the right repository? Open an issue on https://github.com/ros-planning/navigation2
For Bug report or feature requests, please fill out the relevant category below
-->

## Bug report

**Required Info:**

- Operating System:
Ubuntu 16.04.
- Installation type:
source
- Version or commit hash:
master branch
- DDS implementation:
default 
- Client library (if applicable):
rclcpp

#### Steps to reproduce issue
<!-- Detailed instructions on how to reliably reproduce this issue http://sscce.org/
``` code that can be copy-pasted is preferred ``` -->
```
colcon build --symlink-install
```

#### Expected behavior
code builds without error
#### Actual behavior
dwb_plugins fails to build

This is the failure I get when building the navigation2 on the Turtlebot3 joule board:

```
--- stderr: dwb_plugins                                                        
CMakeFiles/twist_gen_test.dir/test/twist_gen.cpp.o: In function `__static_initialization_and_destruction_0(int, int)':
twist_gen.cpp:(.text+0x944f): undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
twist_gen.cpp:(.text+0x949c): undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
twist_gen.cpp:(.text+0x94e9): undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
twist_gen.cpp:(.text+0x9536): undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
twist_gen.cpp:(.text+0x9583): undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
CMakeFiles/twist_gen_test.dir/test/twist_gen.cpp.o:twist_gen.cpp:(.text+0x95d0): more undefined references to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)' follow
collect2: error: ld returned 1 exit status
make[2]: *** [twist_gen_test] Error 1
make[1]: *** [CMakeFiles/twist_gen_test.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
CMakeFiles/goal_checker.dir/test/goal_checker.cpp.o: In function `__static_initialization_and_destruction_0(int, int)':
goal_checker.cpp:(.text+0xbcc): undefined reference to `testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
collect2: error: ld returned 1 exit status
make[2]: *** [goal_checker] Error 1
make[1]: *** [CMakeFiles/goal_checker.dir/all] Error 2
make: *** [all] Error 2
---
Failed   <<< dwb_plugins	[ Exited with code 2 ]
Aborted  <<< nav2_smart_planner

Summary: 21 packages finished [44.7s]
  1 package failed: dwb_plugins
  1 package aborted: nav2_smart_planner
  1 package had stderr output: dwb_plugins
  3 packages not processed
turtlebot@turtlebot:~/navigation2_ws$ 
```
#### Additional information
It looks to me like it's a dependency on gtest, which is where the testing::internal::MakeAndRegisterTestInfo function is.

I have version 1.7 of gtest installed. 
```
turtlebot@turtlebot:~/navigation2_ws$ apt search libgtest Sorting... Done Full Text Search... Done libgtest-dev/xenial,now 1.7.0-4ubuntu1 amd64 [installed]
  Google's framework for writing C++ tests - header files
```

--- a/nav2_dwb_controller/dwb_plugins/CMakeLists.txt
+++ b/nav2_dwb_controller/dwb_plugins/CMakeLists.txt
@@ -54,12 +54,12 @@ if(BUILD_TESTING)
   ament_add_gtest(vtest test/velocity_iterator_test.cpp)
 
   #find_package(rostest REQUIRED)
-  add_executable(goal_checker test/goal_checker.cpp)
-  target_link_libraries(goal_checker simple_goal_checker stopped_goal_checker ${GTEST_LIBRARIES})
+  ament_add_gtest_executable(goal_checker test/goal_checker.cpp)
+  target_link_libraries(goal_checker simple_goal_checker stopped_goal_checker)
   #add_rostest(test/goal_checker.launch DEPENDENCIES goal_checker)
 
-  add_executable(twist_gen_test test/twist_gen.cpp)
-  target_link_libraries(twist_gen_test standard_traj_generator ${GTEST_LIBRARIES})
+  ament_add_gtest_executable(twist_gen_test test/twist_gen.cpp)
+  target_link_libraries(twist_gen_test standard_traj_generator)
   #add_rostest(test/twist_gen.launch DEPENDENCIES twist_gen_test)
 
 endif()

--- a/nav2_dwb_controller/dwb_plugins/CMakeLists.txt
+++ b/nav2_dwb_controller/dwb_plugins/CMakeLists.txt
@@ -51,17 +51,7 @@ if(BUILD_TESTING)
   ament_lint_auto_find_test_dependencies()
 
   find_package(ament_cmake_gtest REQUIRED)
-  ament_add_gtest(vtest test/velocity_iterator_test.cpp)
-
-  #find_package(rostest REQUIRED)
-  ament_add_gtest_executable(goal_checker test/goal_checker.cpp)
-  target_link_libraries(goal_checker simple_goal_checker stopped_goal_checker)
-  #add_rostest(test/goal_checker.launch DEPENDENCIES goal_checker)
-
-  ament_add_gtest_executable(twist_gen_test test/twist_gen.cpp)
-  target_link_libraries(twist_gen_test standard_traj_generator)
-  #add_rostest(test/twist_gen.launch DEPENDENCIES twist_gen_test)
-
+  add_subdirectory(test)
 endif()
 
 install(TARGETS simple_goal_checker stopped_goal_checker standard_traj_generator

--- a/nav2_dwb_controller/dwb_plugins/test/CMakeLists.txt
+++ b/nav2_dwb_controller/dwb_plugins/test/CMakeLists.txt
@@ -1,0 +1,8 @@
+ament_add_gtest(vtest velocity_iterator_test.cpp)
+
+ament_add_gtest(goal_checker goal_checker.cpp)
+target_link_libraries(goal_checker simple_goal_checker stopped_goal_checker)
+
+# Re-enable when DWB reads kinematic parameters
+# ament_add_gtest(twist_gen_test twist_gen.cpp)
+# target_link_libraries(twist_gen_test standard_traj_generator)

--- a/nav2_dwb_controller/dwb_plugins/test/goal_checker.launch
+++ b/nav2_dwb_controller/dwb_plugins/test/goal_checker.launch
@@ -1,3 +1,0 @@
-<launch>
-  <test test-name="goal_checker" pkg="dwb_plugins" type="goal_checker" />
-</launch>

--- a/nav2_dwb_controller/dwb_plugins/test/twist_gen.cpp
+++ b/nav2_dwb_controller/dwb_plugins/test/twist_gen.cpp
@@ -55,7 +55,7 @@ void checkLimits(
   double exp_max_xy = -1.0,
   double exp_min_xy = -1.0, double exp_min_speed_theta = -1.0)
 {
-  ASSERT_GT(twists.size(), 0);
+  ASSERT_GT(twists.size(), 0u);
   nav_2d_msgs::msg::Twist2D first = twists[0];
 
   double min_x = first.x, max_x = first.x, min_y = first.y, max_y = first.y;
@@ -98,7 +98,7 @@ TEST(VelocityIterator, standard_gen)
   StandardTrajectoryGenerator gen;
   gen.initialize(nh);
   std::vector<nav_2d_msgs::msg::Twist2D> twists = gen.getTwists(zero);
-  EXPECT_EQ(twists.size(), 1926);
+  EXPECT_EQ(twists.size(), 1926u);
   checkLimits(twists, 0.0, 0.55, -0.1, 0.1, -1.0, 1.0, 0.55, 0.1, 0.4);
 }
 
@@ -111,7 +111,7 @@ TEST(VelocityIterator, max_xy)
 
   std::vector<nav_2d_msgs::msg::Twist2D> twists = gen.getTwists(zero);
   // Expect more twists since max_speed_xy is now beyond feasible limits
-  EXPECT_EQ(twists.size(), 2010);
+  EXPECT_EQ(twists.size(), 2010u);
   checkLimits(twists, 0.0, 0.55, -0.1, 0.1, -1.0, 1.0, hypot(0.55, 0.1));
 }
 
@@ -123,7 +123,7 @@ TEST(VelocityIterator, min_xy)
   gen.initialize(nh);
   std::vector<nav_2d_msgs::msg::Twist2D> twists = gen.getTwists(zero);
   // Expect even more since theres no min_speed_xy
-  EXPECT_EQ(twists.size(), 2015);
+  EXPECT_EQ(twists.size(), 2015u);
   checkLimits(twists, 0.0, 0.55, -0.1, 0.1, -1.0, 1.0);
 }
 
@@ -135,7 +135,7 @@ TEST(VelocityIterator, min_theta)
   gen.initialize(nh);
   std::vector<nav_2d_msgs::msg::Twist2D> twists = gen.getTwists(zero);
   // Expect even more since theres no min_speed_xy
-  EXPECT_EQ(twists.size(), 2015);
+  EXPECT_EQ(twists.size(), 2015u);
   checkLimits(twists, 0.0, 0.55, -0.1, 0.1, -1.0, 1.0);
 }
 
@@ -149,7 +149,7 @@ TEST(VelocityIterator, no_limits)
   gen.initialize(nh);
   std::vector<nav_2d_msgs::msg::Twist2D> twists = gen.getTwists(zero);
   // vx_samples * vtheta_samples * vy_samples + added zero theta samples - (0,0,0)
-  EXPECT_EQ(twists.size(), 20 * 20 * 5 + 100 - 1);
+  EXPECT_EQ(twists.size(), 20u * 20u * 5u + 100u - 1u);
   checkLimits(twists, 0.0, 0.55, -0.1, 0.1, -1.0, 1.0, hypot(0.55, 0.1), 0.0, 0.0);
 }
 
@@ -166,7 +166,7 @@ TEST(VelocityIterator, no_limits_samples)
   StandardTrajectoryGenerator gen;
   gen.initialize(nh);
   std::vector<nav_2d_msgs::msg::Twist2D> twists = gen.getTwists(zero);
-  EXPECT_EQ(twists.size(), x_samples * y_samples * theta_samples - 1);
+  EXPECT_EQ(twists.size(), static_cast<unsigned>(x_samples * y_samples * theta_samples - 1));
   checkLimits(twists, 0.0, 0.55, -0.1, 0.1, -1.0, 1.0, hypot(0.55, 0.1), 0.0, 0.0);
 }
 
@@ -195,7 +195,7 @@ TEST(VelocityIterator, dwa_gen)
   gen.initialize(nh);
   std::vector<nav_2d_msgs::msg::Twist2D> twists = gen.getTwists(zero);
   // Same as no-limits since everything is within our velocity limits
-  EXPECT_EQ(twists.size(), 20 * 20 * 5 + 100 - 1);
+  EXPECT_EQ(twists.size(), 20u * 20u * 5u + 100u - 1u);
   checkLimits(twists, 0.0, 0.125, -0.1, 0.1, -0.16, 0.16, hypot(0.125, 0.1), 0.0, 0.1);
 }
 
@@ -206,7 +206,7 @@ TEST(VelocityIterator, dwa_gen_no_param)
   dwb_plugins::LimitedAccelGenerator gen;
   gen.initialize(nh);
   std::vector<nav_2d_msgs::msg::Twist2D> twists = gen.getTwists(zero);
-  EXPECT_EQ(twists.size(), 20 * 20 * 5 + 100 - 1);
+  EXPECT_EQ(twists.size(), 20u * 20u * 5u + 100u - 1u);
   checkLimits(twists, 0.0, 0.125, -0.1, 0.1, -0.16, 0.16, hypot(0.125, 0.1), 0.0, 0.1);
 }
 
@@ -222,7 +222,7 @@ TEST(VelocityIterator, nonzero)
   initial.y = -0.08;
   initial.theta = 0.05;
   std::vector<nav_2d_msgs::msg::Twist2D> twists = gen.getTwists(initial);
-  EXPECT_EQ(twists.size(), 2519);
+  EXPECT_EQ(twists.size(), 2519u);
   checkLimits(twists, 0.0, 0.225, -0.1, 0.045, -0.11000000000000003, 0.21,
     0.24622144504490268, 0.0, 0.1);
 }
@@ -362,7 +362,7 @@ TEST(TrajectoryGenerator, accel)
   dwb_msgs::msg::Trajectory2D res = gen.generateTrajectory(origin, zero, forward);
   matchTwist(res.velocity, forward);
   EXPECT_DOUBLE_EQ(durationToSec(res.duration), 5.0);
-  ASSERT_EQ(res.poses.size(), 5);
+  ASSERT_EQ(res.poses.size(), 5u);
   matchPose(res.poses[0], origin);
   matchPose(res.poses[1], 0.1, 0, 0);
   matchPose(res.poses[2], 0.3, 0, 0);
@@ -386,7 +386,7 @@ TEST(TrajectoryGenerator, dwa)
   dwb_msgs::msg::Trajectory2D res = gen.generateTrajectory(origin, zero, forward);
   matchTwist(res.velocity, forward);
   EXPECT_DOUBLE_EQ(durationToSec(res.duration), 5.0);
-  ASSERT_EQ(res.poses.size(), 5);
+  ASSERT_EQ(res.poses.size(), 5u);
   matchPose(res.poses[0], origin);
   matchPose(res.poses[1], 0.3, 0, 0);
   matchPose(res.poses[2], 0.6, 0, 0);

--- a/nav2_dwb_controller/dwb_plugins/test/twist_gen.cpp
+++ b/nav2_dwb_controller/dwb_plugins/test/twist_gen.cpp
@@ -35,6 +35,7 @@
 #include <cmath>
 #include <vector>
 #include <algorithm>
+#include <string>
 #include "gtest/gtest.h"
 #include "dwb_plugins/standard_traj_generator.hpp"
 #include "dwb_plugins/limited_accel_generator.hpp"
@@ -47,6 +48,39 @@ using dwb_plugins::StandardTrajectoryGenerator;
 geometry_msgs::msg::Pose2D origin;
 nav_2d_msgs::msg::Twist2D zero;
 nav_2d_msgs::msg::Twist2D forward;
+
+std::vector<rclcpp::Parameter> getDefaultKinematicParameters()
+{
+  std::vector<rclcpp::Parameter> parameters;
+  parameters.push_back(rclcpp::Parameter("min_vel_x", 0.0));
+  parameters.push_back(rclcpp::Parameter("max_vel_x", 0.55));
+  parameters.push_back(rclcpp::Parameter("min_vel_y", -0.1));
+  parameters.push_back(rclcpp::Parameter("max_vel_y", 0.1));
+  parameters.push_back(rclcpp::Parameter("max_vel_theta", 1.0));
+
+  parameters.push_back(rclcpp::Parameter("acc_lim_x", 2.5));
+  parameters.push_back(rclcpp::Parameter("acc_lim_y", 2.5));
+  parameters.push_back(rclcpp::Parameter("acc_lim_theta", 3.2));
+  parameters.push_back(rclcpp::Parameter("decel_lim_x", -2.5));
+  parameters.push_back(rclcpp::Parameter("decel_lim_y", -2.5));
+  parameters.push_back(rclcpp::Parameter("decel_lim_theta", -3.2));
+
+  parameters.push_back(rclcpp::Parameter("min_speed_xy", 0.1));
+  parameters.push_back(rclcpp::Parameter("max_speed_xy", 0.55));
+  parameters.push_back(rclcpp::Parameter("min_speed_theta", 0.4));
+
+  return parameters;
+}
+
+rclcpp::Node::SharedPtr makeTestNode(const std::string & name)
+{
+  return rclcpp::Node::make_shared(
+    name,
+    "",  // namespace
+    rclcpp::contexts::default_context::get_global_default_context(),
+    std::vector<std::string>(),  // arguments
+    getDefaultKinematicParameters());
+}
 
 void checkLimits(
   const std::vector<nav_2d_msgs::msg::Twist2D> & twists,
@@ -94,7 +128,7 @@ double durationToSec(builtin_interfaces::msg::Duration d)
 
 TEST(VelocityIterator, standard_gen)
 {
-  auto nh = rclcpp::Node::make_shared("st_gen");
+  auto nh = makeTestNode("st_gen");
   StandardTrajectoryGenerator gen;
   gen.initialize(nh);
   std::vector<nav_2d_msgs::msg::Twist2D> twists = gen.getTwists(zero);
@@ -104,7 +138,7 @@ TEST(VelocityIterator, standard_gen)
 
 TEST(VelocityIterator, max_xy)
 {
-  auto nh = rclcpp::Node::make_shared("max_xy");
+  auto nh = makeTestNode("max_xy");
   nh->set_parameters({rclcpp::Parameter("max_speed_xy", 1.0)});
   StandardTrajectoryGenerator gen;
   gen.initialize(nh);
@@ -117,7 +151,7 @@ TEST(VelocityIterator, max_xy)
 
 TEST(VelocityIterator, min_xy)
 {
-  auto nh = rclcpp::Node::make_shared("min_xy");
+  auto nh = makeTestNode("min_xy");
   nh->set_parameters({rclcpp::Parameter("min_speed_xy", -1)});
   StandardTrajectoryGenerator gen;
   gen.initialize(nh);
@@ -129,7 +163,7 @@ TEST(VelocityIterator, min_xy)
 
 TEST(VelocityIterator, min_theta)
 {
-  auto nh = rclcpp::Node::make_shared("min_theta");
+  auto nh = makeTestNode("min_theta");
   nh->set_parameters({rclcpp::Parameter("min_speed_theta", -1)});
   StandardTrajectoryGenerator gen;
   gen.initialize(nh);
@@ -141,7 +175,7 @@ TEST(VelocityIterator, min_theta)
 
 TEST(VelocityIterator, no_limits)
 {
-  auto nh = rclcpp::Node::make_shared("no_limits");
+  auto nh = makeTestNode("no_limits");
   nh->set_parameters({rclcpp::Parameter("max_speed_xy", -1.0)});
   nh->set_parameters({rclcpp::Parameter("min_speed_xy", -1)});
   nh->set_parameters({rclcpp::Parameter("min_speed_theta", -1)});
@@ -155,7 +189,7 @@ TEST(VelocityIterator, no_limits)
 
 TEST(VelocityIterator, no_limits_samples)
 {
-  auto nh = rclcpp::Node::make_shared("no_limits_samples");
+  auto nh = makeTestNode("no_limits_samples");
   nh->set_parameters({rclcpp::Parameter("max_speed_xy", -1.0)});
   nh->set_parameters({rclcpp::Parameter("min_speed_xy", -1)});
   nh->set_parameters({rclcpp::Parameter("min_speed_theta", -1)});
@@ -172,7 +206,7 @@ TEST(VelocityIterator, no_limits_samples)
 
 TEST(VelocityIterator, dwa_gen_exception)
 {
-  auto nh = rclcpp::Node::make_shared("dwa_gen_exception");
+  auto nh = makeTestNode("dwa_gen_exception");
   nh->set_parameters({rclcpp::Parameter("use_dwa", true)});
   StandardTrajectoryGenerator gen;
   EXPECT_THROW(gen.initialize(nh), nav_core2::PlannerException);
@@ -180,7 +214,7 @@ TEST(VelocityIterator, dwa_gen_exception)
 
 TEST(VelocityIterator, no_dwa_gen_exception)
 {
-  auto nh = rclcpp::Node::make_shared("no_dwa_gen_exception");
+  auto nh = makeTestNode("no_dwa_gen_exception");
   nh->set_parameters({rclcpp::Parameter("use_dwa", false)});
   dwb_plugins::LimitedAccelGenerator gen;
   EXPECT_THROW(gen.initialize(nh), nav_core2::PlannerException);
@@ -188,7 +222,7 @@ TEST(VelocityIterator, no_dwa_gen_exception)
 
 TEST(VelocityIterator, dwa_gen)
 {
-  auto nh = rclcpp::Node::make_shared("dwa_gen");
+  auto nh = makeTestNode("dwa_gen");
   nh->set_parameters({rclcpp::Parameter("use_dwa", true)});
   nh->set_parameters({rclcpp::Parameter("min_speed_theta", -1)});
   dwb_plugins::LimitedAccelGenerator gen;
@@ -201,7 +235,7 @@ TEST(VelocityIterator, dwa_gen)
 
 TEST(VelocityIterator, dwa_gen_no_param)
 {
-  auto nh = rclcpp::Node::make_shared("dwa_gen_no_param");
+  auto nh = makeTestNode("dwa_gen_no_param");
   nh->set_parameters({rclcpp::Parameter("min_speed_theta", -1)});
   dwb_plugins::LimitedAccelGenerator gen;
   gen.initialize(nh);
@@ -212,7 +246,7 @@ TEST(VelocityIterator, dwa_gen_no_param)
 
 TEST(VelocityIterator, nonzero)
 {
-  auto nh = rclcpp::Node::make_shared("nonzero");
+  auto nh = makeTestNode("nonzero");
   nh->set_parameters({rclcpp::Parameter("use_dwa", true)});
   nh->set_parameters({rclcpp::Parameter("min_speed_theta", -1)});
   dwb_plugins::LimitedAccelGenerator gen;
@@ -261,7 +295,7 @@ void matchTwist(
 
 TEST(TrajectoryGenerator, basic)
 {
-  auto nh = rclcpp::Node::make_shared("basic");
+  auto nh = makeTestNode("basic");
   StandardTrajectoryGenerator gen;
   gen.initialize(nh);
   dwb_msgs::msg::Trajectory2D res = gen.generateTrajectory(origin, forward, forward);
@@ -277,7 +311,7 @@ TEST(TrajectoryGenerator, basic)
 
 TEST(TrajectoryGenerator, too_slow)
 {
-  auto nh = rclcpp::Node::make_shared("too_slow");
+  auto nh = makeTestNode("too_slow");
   StandardTrajectoryGenerator gen;
   gen.initialize(nh);
   nav_2d_msgs::msg::Twist2D cmd;
@@ -294,7 +328,7 @@ TEST(TrajectoryGenerator, too_slow)
 
 TEST(TrajectoryGenerator, holonomic)
 {
-  auto nh = rclcpp::Node::make_shared("holonomic");
+  auto nh = makeTestNode("holonomic");
   StandardTrajectoryGenerator gen;
   gen.initialize(nh);
   nav_2d_msgs::msg::Twist2D cmd;
@@ -313,7 +347,7 @@ TEST(TrajectoryGenerator, holonomic)
 
 TEST(TrajectoryGenerator, twisty)
 {
-  auto nh = rclcpp::Node::make_shared("twisty");
+  auto nh = makeTestNode("twisty");
   StandardTrajectoryGenerator gen;
   gen.initialize(nh);
   nav_2d_msgs::msg::Twist2D cmd;
@@ -333,7 +367,7 @@ TEST(TrajectoryGenerator, twisty)
 
 TEST(TrajectoryGenerator, sim_time)
 {
-  auto nh = rclcpp::Node::make_shared("sim_time");
+  auto nh = makeTestNode("sim_time");
   nh->set_parameters({rclcpp::Parameter("sim_time", 2.5)});
   StandardTrajectoryGenerator gen;
   gen.initialize(nh);
@@ -350,7 +384,7 @@ TEST(TrajectoryGenerator, sim_time)
 
 TEST(TrajectoryGenerator, accel)
 {
-  auto nh = rclcpp::Node::make_shared("accel");
+  auto nh = makeTestNode("accel");
   nh->set_parameters({rclcpp::Parameter("sim_time", 5.0)});
   nh->set_parameters({rclcpp::Parameter("discretize_by_time", true)});
   nh->set_parameters({rclcpp::Parameter("sim_granularity", 1.0)});
@@ -372,7 +406,7 @@ TEST(TrajectoryGenerator, accel)
 
 TEST(TrajectoryGenerator, dwa)
 {
-  auto nh = rclcpp::Node::make_shared("dwa");
+  auto nh = makeTestNode("dwa");
   nh->set_parameters({rclcpp::Parameter("use_dwa", true)});
   nh->set_parameters({rclcpp::Parameter("sim_period", 1.0)});
   nh->set_parameters({rclcpp::Parameter("sim_time", 5.0)});

--- a/nav2_dwb_controller/dwb_plugins/test/twist_gen.launch
+++ b/nav2_dwb_controller/dwb_plugins/test/twist_gen.launch
@@ -1,3 +1,0 @@
-<launch>
-  <test test-name="twist_gen_test" pkg="dwb_plugins" type="twist_gen_test" />
-</launch>

